### PR TITLE
#48 Calculate v2 of signature  header and update all methods to use it

### PR DESCRIPTION
**Pull request by ernieturner**

### CODE REVIEW

File: src/frame/worker/crypto/recrypt/RecryptWasm.ts [hunk 1]
```diff
@@ -205,22 +211,37 @@ export const decryptPlaintext = (encryptedPlaintext: TransformedEncryptedMessage
     });
 
 /**
- * Create a message signature of the current time, segment ID, user ID, and public signing key. Encode that as a base64 string and sign it
- * using ed25519.
+ * Create our complex request signature using the provided paramters. Details about how this signature works are written up in
+ * https://github.com/IronCoreLabs/design-docs/issues/369
```
Comment (clintfred) [inline src/frame/worker/crypto/recrypt/RecryptWasm.ts]: If you'd rather link to or copy in something from a public repo you could reference https://github.com/IronCoreLabs/ironoxide/issues/50

Comment (ernieturner) [reply src/frame/worker/crypto/recrypt/RecryptWasm.ts]: Oh good call, I wasn't even thinking about that repo being private, thanks!

File: src/frame/endpoints/UserApiEndpoints.ts [hunk 2]
```diff
@@ -139,79 +124,71 @@ const userCreateWithDevice = (keys: UserCreationKeys): RequestMeta => {
...
-function deleteCurrentDevice(userID: string): RequestMeta {
-    return {
-        url: `users/${userID}/devices/current`,
-        options: {
-            method: "DELETE",
-            headers: {
-                "Content-Type": "application/json",
-            },
+const deleteCurrentDevice = (userID: string): RequestMeta => ({
+    url: `users/${userID}/devices/current`,
```
[the remark this answers is not in the document]
Comment (ernieturner) [reply src/frame/endpoints/UserApiEndpoints.ts]: It has not, good catch here. Also fixed the ID in the `userUpdate` URL below.

File: src/frame/worker/crypto/recrypt/RecryptWasm.ts [hunk 3]
```diff
@@ -205,22 +211,37 @@ export const decryptPlaintext = (encryptedPlaintext: TransformedEncryptedMessage
...
+): MessageSignature => {
+    const userContext = [Date.now(), segmentID, userID, fromByteArray(signingKeys.publicKey)].join(",");
+    const userContextBytes = encode(userContext);
+    const authHeaderSignature = RecryptApi.ed25519Sign(signingKeys.privateKey, userContextBytes);
+
+    let bodyAsBytes = new Uint8Array();
+    if (body instanceof Uint8Array) {
+        bodyAsBytes = body;
+    }
+    if (typeof body === "string") {
```
[the remark this answers is not in the document]
Comment (ernieturner) [reply src/frame/worker/crypto/recrypt/RecryptWasm.ts]: Fixed

**Pull request merged**